src/main/java/seedu/address/logic/parser/FindRestaurantCommandParser.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import seedu.address.logic.commands.FindRestaurantCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.restaurant.LocationContainsKeywordsPredicate;
+import seedu.address.model.restaurant.RNameContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindRestaurantCommand object
+ */
+public class FindRestaurantCommandParser implements Parser<FindRestaurantCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindRestaurantCommand
+     * and returns a FindRestaurantCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindRestaurantCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindRestaurantCommand.MESSAGE_USAGE));
+        }
+
+        String[] nameKeywords = trimmedArgs.split("\\s+");
+
+        ArrayList<String> rNameKeywords = new ArrayList<>();
+        ArrayList<String> locationKeywords = new ArrayList<>();
+
+        for (int i = 0; i < nameKeywords.length; i++) {
+            if (nameKeywords[i].contains("n/")) {
+                nameKeywords[i] = nameKeywords[i].substring(2);
+                while (!nameKeywords[i].contains("l/") && i != nameKeywords.length) {
+                    rNameKeywords.add(nameKeywords[i]);
+                    i++;
+                    if (i == nameKeywords.length) {
+                        break;
+                    }
+                }
+                break;
+            }
+        }
+
+        for (int i = 0; i < nameKeywords.length; i++) {
+            if (nameKeywords[i].contains("l/")) {
+                nameKeywords[i] = nameKeywords[i].substring(2);
+                while (!nameKeywords[i].contains("n/") && i != nameKeywords.length) {
+                    locationKeywords.add(nameKeywords[i]);
+                    i++;
+                    if (i == nameKeywords.length) {
+                        break;
+                    }
+                }
+                break;
+            }
+        }
```

Comment:
Not sure if you wanna change it to (rt)find l/LOCATION... n/NAME... instead then can just use the ArgMultiMap thing to get all the keywords for each field O: HAHA

Reply:
Agree, would be easier to parse the keywords!

Reply:
Errr I think it's okay because it would be easier for the user 
Instead of having to write "l/" prefix at the start of every location filter, they can just write it once and then specify all the keywords behind it